### PR TITLE
CTCTRADERS- 578

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/models/MovementReferenceNumber.scala [hunk 1]
```diff
@@ -72,6 +75,11 @@ object MovementReferenceNumber {
       JsString(mrn.toString)
   }
 
+  implicit val xmlReader: XmlReader[MovementReferenceNumber] = {
+    (XmlPath \ "DocNumHEA5").read[String] map (MovementReferenceNumber(_).getOrElse(
+      throw new InvalidPropertiesFormatException("DocNumHEA5: MRN not in right format"))) // TODO needs to handle get
+  }
```
Comment (chris-ianson) [inline app/models/MovementReferenceNumber.scala:81]: We need to have a think about how we handle failed validation of the MRN. If for some reason it's not a valid MRN and has got back to us, should we allow the user to use it in a resubmission? We don't allow the MRN so should we just error out or give them an option to submit anyway? (skip validation)

File: app/services/UnloadingRemarksService.scala [hunk 2]
```diff
@@ -67,4 +69,13 @@ class UnloadingRemarksService @Inject()(config: FrontendAppConfig,
           None
       }
 
+  def resubmit(arrivalId: ArrivalId, registrationNumber: String)(implicit hc: HeaderCarrier): Future[Option[Int]] =
+    unloadingRemarksMessageService.unloadingRemarksMessage(arrivalId) flatMap {
+      case Some(unloadingRemarksXml) =>
+        val updatedXml: NodeSeq = XMLTransformer.updateXmlNode("CorValTOC4", registrationNumber, unloadingRemarksXml)
```
Comment (chris-ianson) [inline app/services/UnloadingRemarksService.scala:75]: The unloading remarks message can have up to 9 _RESOFCON534_ nodes. Within these they each have a _CorValTOC4_ node. Would the updateXmlNode method change all 9 nodes? We may need to target the correct _CorValTOC4_ by finding it by it's _PoiToTheAttTOC5_ (TransportIdentity in this case)

File: app/utils/XMLTransformer.scala [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import scala.xml.transform.{RewriteRule, RuleTransformer}
+import scala.xml.{Elem, Node, NodeSeq, Text}
+
+object XMLTransformer {
+
+  private def createRuleTransformer(key: String, value: String): RuleTransformer =
+    new RuleTransformer(new RewriteRule {
+      override def transform(n: Node): Seq[Node] = n match {
+        case elem: Elem if elem.label.equalsIgnoreCase(key) =>
```
Comment (chris-ianson) [inline app/utils/XMLTransformer.scala:27]: Be interested to know if it's possible to target a specific element (so we can specify the correct _RESOFCON534_ to update)

File: test/models/MovementReferenceNumberSpec.scala [hunk 4]
```diff
@@ -168,5 +166,22 @@ class MovementReferenceNumberSpec extends FreeSpec with MustMatchers with ScalaC
...
+          mrn =>
+            val xml: Node = <HEAHEA>{<DocNumHEA5>{mrn}</DocNumHEA5>}</HEAHEA>
+            val result    = XmlReader.of[MovementReferenceNumber].read(xml)
+            result.toOption.value mustEqual mrn
+        }
+      }
+
+      "must throw an exception if MRN not in right format" in {
+        val xml: Node = <HEAHEA>{<DocNumHEA5>{"mrn"}</DocNumHEA5>}</HEAHEA>
+        val exception = intercept[Exception](XmlReader.of[MovementReferenceNumber].read(xml))
```
Comment (chris-ianson) [inline test/models/MovementReferenceNumberSpec.scala:182]: Mentioned above but need to discuss what to do if MRN is invalid.

File: app/models/MovementReferenceNumber.scala (hunk 1 above)
Comment (ashwinirudrappa) [reply app/models/MovementReferenceNumber.scala:81]: @chris-ianson We should errors out sending invalid MRN is not an acceptable behavior 99.9 it won't happen, worst case if it happens its a hard stop.

File: app/services/UnloadingRemarksService.scala (hunk 2 above)
Comment (ashwinirudrappa) [reply app/services/UnloadingRemarksService.scala:75]: Let's discuss about all possible scenarios

Comment (ashwinirudrappa) [reply app/services/UnloadingRemarksService.scala:75]: @chris-ianson If we want to update multiple fields the good approach is to read XML as a model and update specific fields

Comment (chris-ianson) [reply app/services/UnloadingRemarksService.scala:75]: Yeah, I'm going down that route now. Let's get together after standup

**Pull request merged**